fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/AccessControllerManager.java

```diff
@@ -51,8 +57,21 @@ private CatalogAccessController getAccessControllerOrDefault(String ctl) {
         return ctlToCtlAccessController.getOrDefault(ctl, internalAccessController);
     }
 
-    public void addCatalogAccessControl(String ctl, CatalogAccessController controller) {
-        ctlToCtlAccessController.put(ctl, controller);
+    public void createAccessController(String ctl, String acFactoryClassName, Map<String, String> prop) {
+        Class<?> factoryClazz = null;
+        try {
+            factoryClazz = Class.forName(acFactoryClassName);
+            AccessControllerFactory factory = (AccessControllerFactory) factoryClazz.newInstance();
+            CatalogAccessController accessController = factory.createAccessController(prop);
+            ctlToCtlAccessController.put(ctl, accessController);
+            LOG.info("create access controller {} for catalog {}", ctl, acFactoryClassName);
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        } catch (InstantiationException e) {
+            throw new RuntimeException(e);
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
     }
 
```

Comment:
need a deleteAccessController method

Reply:
fixed. the access controller will be removed when catalog is dropped